traveler-ios-ui/TravelerKitUI/ViewControllers/BookingItemSearch/BookingItemQueryCategoryViewController.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  BookingItemQueryCategoryViewController.swift
+//  TravelerKitUI
+//
+//  Created by Rakin Hoque on 2019-11-15.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import UIKit
+import TravelerKit
+
+public protocol BookingItemQueryCategoryViewControllerDelegate: class {
+    func bookingItemQueryCategoryViewController(_ controller: BookingItemQueryCategoryViewController, didUpdate categories: [ProductItemCategory])
+}
+
+open class BookingItemQueryCategoryViewController: UITableViewController {
+    public weak var delegate: BookingItemQueryCategoryViewControllerDelegate?
+    
+    public var categories: [ProductItemCategory] = []
+    public var selectedCategories: [ProductItemCategory] = []
+    
+    public override func viewDidLoad() {
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: CellIdentifiers.generic)
+        
+        for i in 0 ..< categories.count {
+            if selectedCategories.contains(categories[i]) {
+                let indexPath = IndexPath(row: i, section: 0)
+                tableView.selectRow(at: indexPath, animated: true, scrollPosition: .none)
+                tableView.delegate?.tableView!(tableView, didSelectRowAt: indexPath)
+            }
+        }
+    }
+    
+    public override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath)
+        cell?.accessoryType = .checkmark
+        
+        if selectedCategories.index(of: categories[indexPath.row]) == nil {
```

Comment:
Needs to be invoked in didSelect and didDeselect, or else there is a delay because a row that is selected will invoke didDeselect on the first click and then didSelect on the second click.  I thought about using a dictionary approach, but would require an odd conversion from an array to dictionary and then dictionary back to array to perform the search which seems unnecessary in this situation.

Reply:
I don't think that's the case, didSelect will be invoked with every tap to the row. As such there will be no delay. As a matter of fact, apple handles state change in didSelect https://developer.apple.com/documentation/uikit/uitableviewdelegate/handling_row_selection_in_a_table_view

Reply:
Updated